lib/ansible/plugins/filter/to_nice_json.yml

```diff
@@ -0,0 +1,64 @@
+DOCUMENTATION:
+  name: to_nice_json
+  author: core team
+  version_added: 'historical'
+  short_description: Convert variable to 'nicely formatted' JSON string
+  description:
+    - Converts an Ansible variable into a 'nicely formatted' JSON string representation. This filter functions as a wrapper to the Python ``json.dumps()`` function.
+    - Ansible internally auto-converts JSON strings into variable structures so this plugin is used to force it into a JSON string.
+  options:
+    _input:
+      description: A variable or expression that returns a data structure
+      type: raw
+      required: true
+      format: positional
```

Comment:
The problem with documenting positional parameters is that their order will get lost, as `options` is a dictionary which will get sorted by keys. (At least for the HTML docs.) Maybe it's better to have another top-level key, say `positional_parameters:`, next to `options:`? It should probably be a list of dicts.

Reply:
either that or have `position: x` on each option (but this requires more complex processing), i really dislike `format` so I'll try your suggestion on the next iteration